src/jit/decomposelongs.cpp

```diff
@@ -1137,11 +1148,13 @@ GenTree* DecomposeLongs::DecomposeShift(LIR::Use& use)
                     m_compiler->lvaIncRefCnts(hiCopy);
 
                     hiResult = m_compiler->gtNewOperNode(GT_RSZ, TYP_INT, hiOp1, shiftByHi);
+                    hiResult->gtFlags |= tree->gtFlags & GTF_ALL_EFFECT;
```

Comment:
Everything below here is either a constant or a shift that should be side-effect free except for the helper call node that is created on [line 1333](https://github.com/dotnet/coreclr/pull/7428/files#diff-ba831de5562a37325b7c0f0e47f944ebR1333).